core/src/main/java/cucumber/runtime/RuntimeOptions.java

```diff
@@ -114,11 +115,11 @@ private void parse(List<String> args) {
                 parsedGlue.add(gluePath);
             } else if (arg.equals("--tags") || arg.equals("-t")) {
                 parsedFilters.add(args.remove(0));
-            } else if (arg.equals("--plugin") || arg.equals("-p")) {
-                addPluginName(args.remove(0));
+            } else if (arg.equals("--plugin") || arg.equals("--add-plugin") || arg.equals("-p")) {
```

Comment:
I'm not sure I like renaming the semantics of `-p` to be `--add-plugin` now. This could be a suprise break for people who are currently using `-p` and expect that behavior to be the same.